operator/pkg/operator.go

```diff
@@ -191,9 +283,16 @@ func (o *Operator) Start(ctx context.Context) error {
 // different events enables the smooth operator upgradeability
 
 // Process of handling batches from V2 events:
-func (o *Operator) handleNewBatchLogV2(newBatchLog *servicemanager.ContractAlignedLayerServiceManagerNewBatchV2) {
+func (o *Operator) handleNewBatchLogV2(newBatchLog *servicemanager.ContractAlignedLayerServiceManagerNewBatchV2, batchProcessedChan chan uint32) {
+	var err error
+	defer func() {
+		if err == nil {
+			batchProcessedChan <- uint32(newBatchLog.Raw.BlockNumber)
```

Comment:
Here instead of sending a message to the chan it received as a parameter, it would be best to either call a function that sends the message to the chan, or call a chan stored in the operator struct. I prefer the former to avoid repeting the code that sends a message to a chan.